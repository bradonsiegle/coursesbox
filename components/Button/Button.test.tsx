import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Button } from './Button';

describe('Button test cases', () => {
	it('should render', () => {
		const onClick = jest.fn();
		const { asFragment } = render(<Button onClick={onClick}>Test</Button>);

		expect(asFragment()).toMatchSnapshot();
	});
	it('should call onClick callback', async () => {
		const onClick = jest.fn();
		render(<Button onClick={onClick}>Test</Button>);
		const button = screen.getByRole('button');
		await userEvent.click(button);
		expect(onClick).toHaveBeenCalled();
	});
});
