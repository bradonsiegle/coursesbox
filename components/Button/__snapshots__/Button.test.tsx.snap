// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases should render 1`] = `
<DocumentFragment>
  <button
    class="css-17kvwxi"
    color="primary"
  >
    Test
  </button>
</DocumentFragment>
`;
